xapian-applications/omega/configure.ac

```diff
@@ -246,6 +246,23 @@ AC_SEARCH_LIBS([socketpair], [xnet],
 dnl fnmatch() is used to match ignore pattern.
 AC_CHECK_FUNCS([fnmatch])
 
+OMINDEX_MODULES=""
+
+dnl check if poppler is available.
+PKG_CHECK_MODULES([POPPLER], [poppler-cpp], [
+     save_LIBS=$LIBS
+     LIBS="$LIBS $POPPLER_LIBS"
+     AC_LINK_IFELSE([#include <poppler/cpp/poppler-version.h>
+                     int main(){
+                         poppler::version_string();
+                         return 0;
+                     }],
+                    [AC_DEFINE(HAVE_POPPLER, 1, [Define HAVE_POPPLER if the library is available])
+                     OMINDEX_MODULES="$OMINDEX_MODULES omindex_pdf"])
+     LIBS=$save_LIBS],[echo 'Disabled worker process related to Poppler'])
```

Comment:
You shouldn't use `echo` here.

The correct way to handle this is to use something like `AC_MSG_CHECKING([if libpoppler is usable])` before your test (you probably want it just before the `save_LIBS=...` here, and then `AC_MSG_RESULT([yes])` and `AC_MSG_RESULT([no])` in the _ACTION_IF_TRUE_ and _ACTION_IF_FALSE_ for the `AC_LINK_IFELSE`.

Reply:
Actually, I need to do something if PKG_CHECK_MODULES cannot find poppler-cpp. If I remove this 'echo' and poppler-cpp is not available, PKG_CHECK_MODULES will end the execution with an error for not having found the dependency.
Probably I should remove echo and use something else.
Maybe I can use AC_MSG_CHECKING before PKG_CHECK_MODULES and AC_MSG_RESULT([no]) if PKG_CHECK_MODULES or AC_LINK_IFELSE fail.

Reply:
> Actually, I need to do something if PKG_CHECK_MODULES cannot find poppler-cpp. If I remove this 'echo' and poppler-cpp is not available, PKG_CHECK_MODULES will end the execution with an error for not having found the dependency.
> Probably I should remove echo and use something else.

You should just be able to specify `[ ]` there - then the action is a single space and the default won't be used.  `PKG_CHECK_MODULES` will report that poppler wasn't found for you.

(The `[` and `]` are m4 quotes.)

> Maybe I can use AC_MSG_CHECKING before PKG_CHECK_MODULES and AC_MSG_RESULT([no]) if PKG_CHECK_MODULES or AC_LINK_IFELSE fail.

That won't work well - `PKG_CHECK_MODULES` itself uses `AC_MSG_CHECKING` and `AC_MSG_RESULT` around its check, so the messages will appear interleaved.

Reply:
I didn't realize of that. I remove the echo message.